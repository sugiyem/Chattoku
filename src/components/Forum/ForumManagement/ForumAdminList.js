import { useEffect, useState } from "react";
import { SectionList } from "react-native";
import { getAllAdmins } from "../../../services/Forum/HandleForumAdmin";
import ForumAdminCard from "./ForumAdminCard";

const ForumAdminList = ({ forumId }) => {
  const [users, setUsers] = useState([]);

  useEffect(() => {
    return getAllAdmins(forumId, setUsers);
  }, []);

  const renderItem = ({ item }) => {
    return <ForumAdminCard userId={item.uid} authorities={item.authorities} />;
  };

  const renderHeader = () => <></>;

  const renderFooter = () => <></>;

  return (
    <SectionList
      removeClippedSubviews={true}
      sections={[
        {
          data: users
        }
      ]}
      keyExtractor={(item) => item.uid}
      renderItem={renderItem}
      renderHeader={renderHeader}
      renderFooter={renderFooter}
    />
  );
};

export default ForumAdminList;
