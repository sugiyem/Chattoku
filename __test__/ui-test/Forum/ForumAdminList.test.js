import React from "react";
import renderer from "react-test-renderer";
import ForumAdminList from "../../../src/components/Forum/ForumManagement/ForumAdminList";

jest.mock("../../../src/services/Profile/FetchUserInfo");
jest.mock("../../../src/services/Forum/HandleForumAdmin");

describe("Test Forum Admin List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<ForumAdminList forumId="forum-1" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
