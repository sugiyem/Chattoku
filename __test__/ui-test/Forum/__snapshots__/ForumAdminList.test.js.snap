// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Forum Admin List UI Renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [],
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
